### PR TITLE
Setup testing

### PR DESCRIPTION
**Pull request by surchs**
- some basic unit and functional tests
- not all tests are implemented yet (TODO tags where not implemented)
- dependencies and config files for running pytest suite
- should be enough to get a CI system running

### CODE REVIEW

**Comment by surchs** [comment]
OK, this is very exciting. The functional tests did in fact fail for two reasons. One of them was good:
1. I had forgotten to (commit) give the functional test database access via the [pytest-django database decorator](https://pytest-django.readthedocs.io/en/latest/database.html) (not a good reason)
2. When I had migrated the graph to our new compute canada cloud instance in https://github.com/metaneuro/project/issues/5, I had chosen a new name for the database (`integration_layer` rather than `demo_integration`). Of course I had forgotten to update the database name in the configuration files. But: I didn't notice this because I was only testing the actual app, not running the functional tests. The actual app still had the cached query responses from before I had moved the database and so happily showed me all of my old results without crashing. By contrast, the functional tests get initialized with a fresh database every time (that gets destroyed afterwards) and so they immediately ran into the issue: `Exception: Query request unsuccessful (404): {"message":"Database 'demo_integraion' does not exist.","code":"0D0DU2"}`

So, bottom line: first time the tests saved us :tada: I would have happily walked into the demo tomorrow, changed something in the query and then scratched my head for half an hour why suddendly nothing works anymore. Very nice

**Comment by jarmoza** [comment]
That's great – and great! Seems like we're ready to merge this PR now.
